src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java

```diff
@@ -25,20 +29,35 @@ public UserController(final UserService service) {
     }
 
     @Override
-    @GetMapping
+    @GetMapping(
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
     public ResponseEntity<List<User>> list() {
         return ResponseEntity.ok(service.findAll());
     }
 
     @Override
-    @GetMapping
-    @RequestMapping("{id}")
+    @GetMapping(
+            path = "/{id}",
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
     public ResponseEntity<User> get(final @PathVariable Long id) {
         return ResponseEntity.ok(service.findById(id));
     }
 
+    @GetMapping(
+            params = "username",
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<User> get(final @RequestParam("username") String username) {
```

Comment:
Since you are finding an individual user, shouldn't this be a path parameter rather than a query parameter?